Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -156,6 +156,8 @@ extension StripeCardReaderService: CardReaderService {
             }
 
             Terminal.shared.collectPaymentMethod(activeIntent, delegate: self) { (intent, error) in
+                self?.sendReaderEvent(CardReaderEvent(type: .cardRemoved))
```

Comment:
Q: Is it ok to send the event before checking for the error or intent?

Reply:
In this case (and only this case), it does. This is the event that notifies the UI that the card has been tapped / swiped. Internally, that means that the operation to collect a payment method has already a response from the SDK (which then can be either an error or success)